drv/lpc55-swd/src/main.rs

```diff
@@ -560,15 +584,15 @@ impl idl::InOrderSpCtrlImpl for ServerImpl {
                         break;
                     }
                 }
-                Err(_) => {
-                    return Err(SpCtrlError::Fault.into());
+                Err(e) => {
+                    return Err(e.into());
```

Comment:
it occurs to me that now that the error case here is just to return `e.into()`, this whole loop could be simplified to just:
```rust
        loop {
            let dhcsr = self.dp_read_bitflags::<Dhcsr>()?;
            ringbuf_entry!(Trace::Dhcsr(dhcsr));
            if dhcsr.is_regrdy() {
                break;
            }
        }
```
but, that's a bigger diff than the previous code, so...up to you whether it's worth the change.